py_hamt/hamt.py

```
@@ -1,674 +1,630 @@
-        Returns:
-            Hamt: Created and saved HAMT
-        """
-        new_node = cls(store, options, map, depth, data)
-        return save(store, new_node)
+    def deserialize(cls, data: bytes) -> "Node":
+        try:
+            decoded = dag_cbor.decode(data)
+        except:  # noqa: E722
+            raise Exception(
+                "Invalid dag-cbor encoded data from the store was attempted to be decoded"
+            )
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)